### ISSUE TITLE
Compile fails with libmicrohttpd v0.9.71

### ISSUE THREAD

**Issue by ilf-S** [jpmens/revgeod#4]
Compile fails on Fedora 33 with libmicrohttpd 0.9.71.

```
>  make
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"    -c -o json.o json.c
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"    -c -o geohash.o geohash.c
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"    -c -o geo.o geo.c
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"    -c -o db.o db.c
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"    -c -o uptime.o uptime.c
cc -Wall -Werror  -DLMDB_DATABASE=\""data/geocache/"\" -DLISTEN_HOST=\""127.0.0.1"\" -DLISTEN_PORT=\""8865"\" -DSTATSD=\""127.0.0.1"\"  -o revgeod revgeod.c json.o geohash.o geo.o db.o uptime.o  statsd/statsd-client.o -lmicrohttpd -lcurl -llmdb 
revgeod.c: In function ‘main’:
revgeod.c:587:3: error: passing argument 5 of ‘MHD_start_daemon’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  587 |   &handle_connection, NULL,
      |   ^~~~~~~~~~~~~~~~~~
      |   |
      |   int (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, size_t *, void **) {aka int (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, long unsigned int *, void **)}
In file included from revgeod.c:37:
/usr/include/microhttpd.h:2428:45: note: expected ‘MHD_AccessHandlerCallback’ {aka ‘enum MHD_Result (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, long unsigned int *, void **)’} but argument is of type ‘int (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, size_t *, void **)’ {aka ‘int (*)(void *, struct MHD_Connection *, const char *, const char *, const char *, const char *, long unsigned int *, void **)’}
 2428 |                   MHD_AccessHandlerCallback dh, void *dh_cls,
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~^~
cc1: all warnings being treated as errors
make: *** [Makefile:23: revgeod] Error 1
```

Googling around seems to point to a change in libmicrohttpd, but I'm not entirely sure.

**Comment by jpmens** [comment]
Further [information here](https://gitlab.com/jeanfi/psensor/-/issues/24)

**Comment by jpmens** [comment]
@ilf-S thanks for your report. I've applied (but not tested) a fix. Could you see if that silences the compiler warning?

**Comment by ckrey** [comment]
To become compatible with future versions extent the fix in 40245fd according to
[Changelog for libmicrohttpd 0.9.71](https://abi-laboratory.pro/index.php?view=changelog&l=libmicrohttpd&v=0.9.71)

**Comment by jpmens** [comment]
thank you, @ckrey
